PatientAnalytics/Services/DatabasePopulateService.cs

```diff
@@ -0,0 +1,61 @@
+using Microsoft.EntityFrameworkCore;
+using PatientAnalytics.Models;
+using PatientAnalytics.Models.Auth;
+using PatientAnalytics.Utils;
+
+namespace PatientAnalytics.Services;
+
+public class DatabasePopulateService(IServiceScopeFactory scopeFactory, IConfiguration configuration) : IHostedService
+{
+    public async Task StartAsync(CancellationToken stoppingToken)
+    {
+        using var scope = scopeFactory.CreateScope();
+
+        await using var context = scope.ServiceProvider.GetRequiredService<Context>();
+
+        await context.Database.EnsureCreatedAsync(stoppingToken);
+
+        var isMigrationNeeded = (await context.Database.GetPendingMigrationsAsync(stoppingToken)).Any();
+
+        if (isMigrationNeeded)
+        {
+            await context.Database.MigrateAsync(stoppingToken);
+        }
+
+        if (!context.Users.Any(u => u.Role == "SuperAdmin"))
+        {
+            await CreateInitialSuperAdmin(context);
+        }
+    }
+
+    private async Task CreateInitialSuperAdmin(Context context)
+    {
+        const string username = "superadmin";
+        var password = Password.GeneratePassword();
+        var hashed = Password.HashPassword(password, configuration);
```

Comment:
suggestion: I’d recommend passing the password as an environment variable / putting it in a vault and loading it from IConfiguration. It makes it a bit easier as you won’t need to retrieve the password after first run

worth noting that the PasswordHasher also uses Random, which provides predictable output, which means somebody could reverse engineer the password that was generated

Reply:
Okay, I will add this as a TODO comment, one of my next piece of work is to remove the secrets from appsettings and inject them as environment variables.